src/dawn/Compiler/DawnCompiler.cpp

```diff
@@ -190,6 +191,10 @@ std::unique_ptr<OptimizerContext> DawnCompiler::runOptimizer(std::shared_ptr<SIR
   optimizer->checkAndPushBack<PassSetBlockSize>();
   optimizer->checkAndPushBack<PassDataLocalityMetric>();
   optimizer->checkAndPushBack<PassSetSyncStage>();
+  // If we want to generate cuda code or serialize the iir we inline all function calls
+  optimizer->checkAndPushBack<PassInlining>(
+      getOptions().Backend == "cuda" || getOptions().SerializeIIR,
```

Comment:
For reference improve comment to include why cuda and serialization needs the inlining.